### DISCUSSION TITLE
How do I select another Java version in my macos workflow?

### DISCUSSION THREAD

**Discussion by notlaein** [actions/virtual-environments#6264] [General questions] [answered]
Hi, I see that the macos images contain Java version 8, 11 and 17, and that 8 is default. There is also an environment variable per version, but I cannot find out how to change to use the different versions.

I tried this, based on guesswork, but it doesnt work:
```
env:
      JAVA_HOME: "${{ JAVA_HOME_11_X64 }}"
```

Guess this is quite basic, but I cannot seem to figure it out on my own...

**Comment by notlaein** [comment, marked as answer]
Tried out some more variations, and stumbled across the solution my self:
```
env:
           JAVA_HOME: JAVA_HOME_11_X64
```

**Reply by miketimofeev** [reply]
@einar-notland those java versions are intended to use with:
- https://github.com/actions/setup-java action in GitHub
- [Java Tool Installer task](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/java-tool-installer?view=azure-devops) in Azure DevOps

**Reply by notlaein** [reply]
Hmm, ok. 
Setting JAVA_HOME seemed to be enough in my case, Im just using java to execute a .jar file.
